app/public/src/pages/game.tsx

```diff
@@ -292,6 +294,24 @@ export default function Game() {
         gameContainer.onDragDropCombine(event)
       }) as EventListener)
 
+      room.onLeave((code) => {
+        const shouldReconnect = code === CloseCodes.ABNORMAL_CLOSURE || code === CloseCodes.TIMEOUT
+        logger.info(`left game room with code ${code}`, { shouldReconnect })
+        if (shouldReconnect) {
+          logger.log("Connection closed unexpectedly or timed out. Attempting reconnect.")
+          // Clear out all game data to reload the game
+          connected.current = false
+          initialized.current = false
+          if (gameContainer && gameContainer.game) {
+            gameContainer.game.destroy(true)
+          }
+          setLoaded(false)
+          dispatch(leaveGame())
+          // Update dependency variable to force a re-render
+          setReconnectTries(x => x + 1)
```

Comment:
you don't need to re-render. There's nothing to update in the component render, it's just reconnecting to room